lib/proto.js

```diff
@@ -193,6 +198,10 @@ function write (_obj, msg, num) {
     if (msg === undefined) {
       msg = _obj.message
     }
+  } else if (_obj instanceof IncomingMessage) {
+    obj = { [requestKey]: _obj }
+  } else if (_obj instanceof ServerResponse) {
+    obj = { [responseKey]: _obj }
```

Comment:
If preferred, we can assess _type_ from properties, as previously done in `lib/tools.js`:
```js
  } else if (_obj.method && _obj.headers && _obj.socket) {
    obj = { [requestKey]: _obj }
  } else if (typeof _obj.setHeader === 'function') {
    obj = { [responseKey]: _obj }
```

Reply:
I think property presence will be better here. These _could_ be extended objects, e.g. a `FastifyRequest`.

Reply:
done